swap/src/protocol/bob/swap.rs

```diff
@@ -103,14 +107,25 @@ async fn next_state(
 
                 select! {
                     transfer_proof = transfer_proof_watcher => {
-                        let transfer_proof = transfer_proof?.tx_lock_proof;
-
-                        tracing::info!(txid = %transfer_proof.tx_hash(), "Alice locked Monero");
-
-                        BobState::XmrLockProofReceived {
-                            state: state3,
-                            lock_transfer_proof: transfer_proof,
-                            monero_wallet_restore_blockheight
+                        let transfer_proof = transfer_proof?;
+
+                        if transfer_proof.swap_id == swap_id {
```

Comment:
I think similar to Alice, we shouldn't even be passing on the message through the channel if it is not for the right swap.